src/webview/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java

```diff
@@ -231,6 +239,27 @@ protected void onStart() {
 							}
 						}
 						return;
+
+					case RDTOOLKIT_PROVISION_ACTIVITY_REQUEST_CODE:
+						String provisionScript = rdToolkitSupport.process(requestCode, resultCode, i);
+						log(this, "RDToolkitSupport :: Executing JavaScript: %s", provisionScript);
+						evaluateJavascript(provisionScript);
+						return;
+
+					case RDTOOLKIT_CAPTURE_ACTIVITY_REQUEST_CODE:
+						int readPermission = ContextCompat.checkSelfPermission(this, READ_EXTERNAL_STORAGE);
+
+						if (readPermission != PERMISSION_GRANTED) {
+							log(this, "RDToolkitSupport :: Requesting storage permissions");
+							ActivityCompat.requestPermissions(this, PERMISSIONS_STORAGE, RDTOOLKIT_CAPTURE_ACTIVITY_REQUEST_CODE);
```

Comment:
This respects the "prominent disclosure" rules, right?

Reply:
Also, is this an async or a sync operation? Will the thread "halt" until the user has granted permission, or will the capture script that follows run anyway (and fail)?

Reply:
> This respects the "prominent disclosure" rules, right?

This permission is already in use in medic-android, but as I read, this should be fine. They did put additional restrictions for MANAGE_EXTERNAL_STORAGE permission (which we still don't require) in Android 11+. 

> Also, is this an async or a sync operation? Will the thread "halt" until the user has granted permission, or will the capture script that follows run anyway (and fail)?

I'm using Android 11.
If the user remove the permission in the phone settings, then when capturing an RD Test result the phone displays a modal asking for granting the permission again.
if the user close the modal (not granting, not rejecting) the results aren't capture but user can continue using the app normally, it's not breaking the app. 
if the user reject the permission, the results aren't capture and the user can use the app normally, it's not breaking the app neither.